internal/server/event_handlers.go

```diff
@@ -251,19 +253,55 @@ func (s *EventServer) getNotificationParams(ctx context.Context, event *eventv1.
 		return nil, nil, "", 0, nil
 	}
 
-	sender, token, err := createNotifier(ctx, s.kubeClient, provider)
+	// Create a copy of the event and combine event metadata
+	notification := *event.DeepCopy()
+	s.combineEventMetadata(ctx, &notification, alert)
+
+	// Create a commit status for the given provider and event, if applicable.
+	commitStatus, err := createCommitStatus(ctx, provider, &notification, alert)
 	if err != nil {
-		return nil, nil, "", 0, fmt.Errorf("failed to initialize notifier for provider '%s': %w", provider.Name, err)
+		logger.Error(err, "failed to evaluate commitStatusExpr", "expression", provider.Spec.CommitStatusExpr, "provider", provider.Name)
+		s.Eventf(&provider, corev1.EventTypeWarning, "CommitStatusExprEvalFailed", "failed to evaluate commitStatusExpr %q for provider %s: %s", provider.Spec.CommitStatusExpr, provider.Name, err)
```

Comment:
We don't need to log/send a k8s event here, just bubbling up the error up to the handler for loop for each Alert matching the event is enough, we log errors and send k8s events there